### ISSUE TITLE
Logo and text is missing in the pdf version of the delivery report

### ISSUE THREAD

**Issue by moahaegglund** [Clinical-Genomics/scout#3108]
## Description

When downloading the pdf version of the delivery report all information doesn't fit, the Swedac logo and "Svarsdatum" is not visible.

From the pdf:
<img width="907" alt="Screenshot 2022-01-25 at 10 05 46" src="https://user-images.githubusercontent.com/19236050/150946141-e304ae50-03be-4340-82ff-aea6d85656c9.png">

From the web link:
<img width="1121" alt="Screenshot 2022-01-25 at 10 07 09" src="https://user-images.githubusercontent.com/19236050/150946208-83309384-add4-4c02-9379-ff0278d59284.png">

## Suggested solution

*Write here if there is a suggested way to solve the issue*

## This can be closed when

All information is visible in the pdf version of the delivery report.

**Comment by patrikgrenfeldt** [comment]
@moahaegglund I transferred this issue to Scout repository since I think the PDF-creation happens in the Scout application

**Comment by northwestwitch** [comment]
Thanks, We'll fix, might be related to this: https://github.com/Clinical-Genomics/chanjo-report/issues/24, but I'm not sure

**Comment by dnil** [comment]
Yep! Prime candidates are either an update to the pdf generation library or possibly a formatting change for the CG generated report (compare the cancer delivery report, where we were able to fix the pdf conversion by changing a few html tags).

**Comment by patrikgrenfeldt** [comment]
> Yep! Prime candidates are either an update to the pdf generation library or possibly a formatting change for the CG generated report (compare the cancer delivery report, where we were able to fix the pdf conversion by changing a few html tags).

Ok, transfer it back if it belongs in cg, please add to the issue what was needed to fix the issue in that case

**Comment by dnil** [comment]
We can take a look and compare to the cancer delivery report changes. It would feel very nice to swap out the conversion library, but last we looked this still was state of the art... 🙄

**Comment by northwestwitch** [comment]
Since the HTML reports are fine, I think the problem is actually in Scout, check where the PDF reports are created in the code: https://github.com/Clinical-Genomics/scout/blob/163c45d4c4dd69abbf74c38baa025eeadf2596c9/scout/server/blueprints/cases/views.py#L684

**Comment by northwestwitch** [comment]
I can take a look at this issue now!

**Comment by northwestwitch** [comment]
Ok looks like the logo was not included in the PDF file in the first place, for rendering issues: https://github.com/Clinical-Genomics/scout/blob/163c45d4c4dd69abbf74c38baa025eeadf2596c9/scout/server/blueprints/cases/views.py#L688

**Comment by moahaegglund** [comment]
Thanks for looking at this @northwestwitch!
@AnnaLeinfelt what do you say about the logo and the pdf version?

**Comment by northwestwitch** [comment]
The missing text due to the weird layout is still to be fixed, I'll get back to you when I understand a little more!

**Comment by dnil** [comment]
Hm, do we really need to keep weasyprint frozen anymore now that we moved away from clinicaldb? Might be worth testing relaxing that freeze? It seems to be under active development again, with v54.0 recently out.

**Comment by northwestwitch** [comment]
> Hm, do we really need to keep weasyprint frozen anymore now that we moved away from clinicaldb? Might be worth testing relaxing that freeze? It seems to be under active development again, with v54.0 recently out.

If we unlock it it might mess up pretty bad with the general report. 

Mmm, this issue reminds me of this one: https://github.com/Clinical-Genomics/BALSAMIC/pull/595/files

**Comment by northwestwitch** [comment]
I could reproduce the content that doesn't fit the PDF page locally using a delivery report from a recent case. 

Comparing with an old report (on the left), it seems that new reports (on the right) contain some more columns on both tables:

![image](https://user-images.githubusercontent.com/28093618/151139820-f0e07e09-8d8f-4089-b38c-09f76e079542.png)

![image](https://user-images.githubusercontent.com/28093618/151140239-e8b57b7c-7900-449e-b622-738cedfd5144.png)

And the new content doesn't fit any more in a A4 page.

I'll get back with a fix soon!

**Comment by AnnaLeinfelt** [comment]
I would prefer to have the log  on the top right corner, or somewhere else where it is nice and neat.

You have all the logos in a folder on google drive [Swedac logos](https://drive.google.com/drive/folders/0BzmgPBcGtZwaTzJNVnh3X3ZnUzA?resourcekey=0-x9YFRFNCCwxsfgM9qctdkQ)

Maybe the smaller logo (49 kb) would fit better?

**Comment by moahaegglund** [comment]
We are discussing what to include in the delivery report for balsamic, [https://github.com/Clinical-Genomics/cg/issues/1378](https://github.com/Clinical-Genomics/cg/issues/1378), it's possible that it will have even more columns.

**Comment by northwestwitch** [comment]
> We are discussing what to include in the delivery report for balsamic, [Clinical-Genomics/cg#1378](https://github.com/Clinical-Genomics/cg/issues/1378), it's possible that it will have even more columns.

Ok then perhaps landscape orientation should be preferred

**Comment by dnil** [comment]
I believe the balsamic delivery report is already a landscape one, so is potentially already wider but could possibly hold more. Note that the reports as such are not generated in Scout, so they will have to be changed in CG (for MIP) or Balsamic!

**Comment by moahaegglund** [comment]
Yes the "Coverage and QC report" for panel cases in balsamic is in landscape format, I'm not sure if we will keep this report or replace it with the delivery report that is about to be added. I see now that there is actually a problem with columns not fitting in the pdf version of this report too: [https://scout.scilifelab.se/cust000/NEQAS-somaticWES](https://scout.scilifelab.se/cust000/NEQAS-somaticWES ).

**Comment by northwestwitch** [comment]
I've created a PR to fix the problem of the missing text in cg now: https://github.com/Clinical-Genomics/cg/pull/1382

**Comment by northwestwitch** [comment]
Moving this issue back to cg, where I've opened a PR to fix it!
